src/offers/mutations.js

```diff
@@ -0,0 +1,106 @@
+import { useMutation, useQueryClient } from 'vue-query'
+import api from '@/offers/api/offers'
+import { unref } from 'vue'
+import { queryKeyOfferDetail, queryKeyOfferList } from '@/offers/queries'
+import { useRoute, useRouter } from 'vue-router'
+import { withStatus } from '@/utils/queryHelpers'
+import { useCurrentGroupIdRef } from '@/group/queries'
+
+// Store the ids we are currently mutating, so we can better decide when to refresh from websocket
+const mutatingOfferIds = new Set()
```

Comment:
Not sure I fully understand - I guess it shouldn't invalidate the offer while the user is editing it? Or is it about some data race when clicking the "save"/"create" button?

Reply:
It's to avoid extra unneeded queries when receiving websocket updates. Without this logic the websocket update will invalidate the offers query and reload the data, and then the response to the mutation request will also do that, and two requests to the backend will go out. I think some situations like that would get debounced, but in this case it would make two separate requests.

Another option is to rely on the websocket update alone, but I tend to not like to rely on that approach, although I think currently we do in places and not others (depending on who wrote the code :), so maybe we should actually decide on a policy there).

I also looked at just updating the specific offer that is saved/edited/archived in the query (using `setQueryData`), but for the query it's not always obvious how a given edit might impact all the queries (e.g. when you archive it, it needs to be added to the status=archived query and removed from the status=active one), so I think just invalidating the lot is better.